runtime/thunk/thunk.ts

```diff
@@ -85,7 +63,9 @@ export function findSyncFormula(packDef: PackVersionDefinition, syncFormulaName:
 
   for (const syncTable of packDef.syncTables) {
     const syncFormula = syncTable.getter;
-    if (syncFormula.name === syncFormulaName) {
+    // TODO(huayang): it seems like a bug that client passes syncTable.name in, instead of syncTable.identityName or
+    // syncTable.getter.name.
+    if (syncTable.name === syncFormulaName) {
```

Comment:
Ok I'm wrong. We seem to consistently use the sync table name everywhere to locate the formula.

The thing I was remembering is that for executing parameter autocomplete, we need to locate the parent formula of the parameter, and I think what the client is passing in that case is the getter name (because the getter is the immediate parent of the parameter):

https://github.com/coda/coda/blob/main/modules/server/packs/v2/runtime/thunk/thunk.ts#L175

Reply:
Is it just temporary that findSyncFormula() or anything else from the thunk is exported?

It seems like the ideal would be to only export findAndExecutePackFunction(), and the simulator in execution.ts would just create a FormulaSpecification and invoke that with whatever it got on the commandline and an appropriate execution context?

Reply:
> Is it just temporary that findSyncFormula() or anything else from the thunk is exported?

Well I can move findSyncFormula() to `runtime/common/` if that's necessary. 

Other helpers like `handleError` or `marshalValue` are necessary to be exported. They will be used in injections. 

```
  await context.evalClosure(
    `${stubName} = async function(...args) {
        return coda.handleErrorAsync(async () => {
         const result = await $0.apply(
           undefined,
           args.map(coda.marshalValue),
           {
             arguments: {copy: true},
             result: {copy: true, promise: true},
           },
         );
         return coda.unmarshalValue(result);
       });
     };`,
    [stub],
    {arguments: {reference: true}},
  );
```

Alternatively we can expose a function builder instead of all the specific methods. Something like 

```
export function buildStubFunctionAsync(func) {
  return async function(...args) {
        return handleErrorAsync(async () => {
         const result = await func.apply(
           undefined,
           args.map(marshalValue),
           {
             arguments: {copy: true},
             result: {copy: true, promise: true},
           },
         );
         return unmarshalValue(result);
       });
     };
  
}
```

so that the injection can be something like

```
await context.evalClosure(
    `${stubName} = coda.buildStubFunctionAsync($0);`,
    [stub],
    {arguments: {reference: true}},
```

This might be a better way actually to avoid in-string code. but anyway something extra needs to be exported for IVM context injections.

Reply:
actually the alternative I proposed will require to install the thunk bundle before injecting globals (e.g. console.log). We may or may not want to do this.

Reply:
No need to move things around, was just trying to understand the usage specifically around findSyncFormula(). The thunk is designed to handle running all of our formula types, so just wasn't sure why we were making a special case of locating and running a sync formula. Thought maybe this is just a temporary thing and this will change to using the main entrypoint of the thunk?

Reply:
these methods will be exported so that they can be used in ivm eval's. if not thunk, then another bundle. preferred to be in one bundle.

Reply:
Oh it's fine to export helpers like `marshalValue()`. 

I was specifically wondering why `findSyncFormula()` needs to be exported separately. Isn't there a world in which we can reuse `findAndExecutePackFunction()` for running syncs in the CLI simulator, too?

Reply:
ah, I can move `findSyncFormula` to common for now. 

> Isn't there a world in which we can reuse findAndExecutePackFunction() for running syncs in the CLI simulator, too?

In an idea world yes. However right now I need to use it to figure out if CLI is actually passing a formula or sync formula. The syntax isn't specifying the formula type.